crates/forge-runner/src/running.rs

```diff
@@ -264,44 +318,87 @@ pub fn run_test_case(
-                    .expect("Execution resources missing")
-                    .filter_unused_builtins();
-                add_resources_to_top_call(
-                    &mut forge_runtime,
-                    &vm_resources_without_inner_calls,
-                    &tracked_resource,
-                );
-
-                let ap = runner.relocated_trace.as_ref().unwrap().last().unwrap().ap;
-
-                let results_data = get_results_data(
-                    &case.test_details.return_types,
-                    &runner.relocated_memory,
-                    ap,
-                );
-                assert_eq!(results_data.len(), 1);
-
-                let (_, values) = results_data[0].clone();
-                let value = SierraCasmRunner::handle_main_return_value(
-                    // Here we assume that all test either panic or do not return any value
-                    // This is true for all test right now, but in case it changes
-                    // this logic will need to be updated
-                    Some(0),
-                    values,
-                    &runner.relocated_memory,
-                );
-
-                update_top_call_vm_trace(&mut forge_runtime, &mut runner);
+    let entry_point_initial_budget = forge_runtime
+        .extended_runtime
+        .extended_runtime
+        .extended_runtime
+        .hint_handler
+        .base
+        .context
+        .gas_costs()
+        .base
+        .entry_point_initial_budget;
+    let args = prepare_call_arguments(
+        &forge_runtime
+            .extended_runtime
+            .extended_runtime
+            .extended_runtime
+            .hint_handler
+            .base
+            .call
+            .clone()
+            .into(),
+        &mut runner,
+        initial_syscall_ptr,
+        &mut forge_runtime
+            .extended_runtime
+            .extended_runtime
+            .extended_runtime
+            .hint_handler
+            .read_only_segments,
+        &entry_point,
+        entry_point_initial_budget,
+    )?;
 
-                Ok(value)
+    let n_total_args = args.len();
+
+    // Execute.
+    let bytecode_length = program.data_len();
+    let program_segment_size = bytecode_length + program_extra_data_length;
+    let result: Result<CallInfo, CairoRunError> = match run_entry_point(
+        &mut runner,
+        &mut forge_runtime,
+        entry_point,
+        args,
+        program_segment_size,
+    ) {
+        Ok(()) => {
+            let call_info = finalize_execution(
+                &mut runner,
+                &mut forge_runtime
+                    .extended_runtime
+                    .extended_runtime
+                    .extended_runtime
+                    .hint_handler,
+                n_total_args,
+                program_extra_data_length,
+                tracked_resource,
+            )?;
+
+            // TODO this can be done better, we can take gas directly from call info
```

Comment:
Will it be addressed in this PR (or issue will be added)?

Reply:
I want to do it in the separate PR. Linked https://github.com/foundry-rs/starknet-foundry/issues/3292